src/core/components/AuthenticateButton/index.js

```diff
@@ -98,15 +99,12 @@ export class AuthenticateButtonBase extends React.Component<InternalProps> {
 }
 
 type StateMappedProps = {|
-  api: ApiStateType,
+  api: ApiState,
   handleLogIn: HandleLogInFunc,
   siteUser: UserType | null,
 |};
 
-export const mapStateToProps = (state: {|
-  api: ApiStateType,
-  users: UsersStateType,
-|}): StateMappedProps => ({
+export const mapStateToProps = (state: AppState): StateMappedProps => ({
```

Comment:
We don't seem to specify a return type for `mapStateToProps` anywhere else, so why are we doing it here? Perhaps this should be removed, or perhaps an issue should be filed to do this for all instances of `mapStateToProps`?

Reply:
That was there before... I am not sure we need to type hint the return value since it is never used directly.

Reply:
It seems odd that you'd get an error though. 🤷‍♂️